src/Magento/FunctionalTestingFramework/Util/Validation/NameValidationUtil.php

```diff
@@ -50,4 +68,246 @@ public static function validateName($name, $type)
             throw new XmlException($errorMessage);
         }
     }
+
+    /**
+     * Validates data entity names.
+     *
+     * @param string $str
+     * @param string $filename
+     * @throws Exception
+     * @return void
+     */
+    public function validateDataEntityName($str, $filename = null)
+    {
+        if (!ctype_upper($str[0])) {
+            $message = "The data entity name {$str} should be PascalCase with an uppercase first letter.";
+
+            if ($filename !== null) {
+                $message .= " See file {$filename}.";
+            }
+
+            LoggingUtil::getInstance()->getLogger(self::class)->notification(
+                $message,
+                [],
+                false
+            );
+
+            $this->count++;
+        }
+    }
+
+    /**
+     * Validates data entity key names.
+     *
+     * @param string $str
+     * @param string $filename
+     * @throws Exception
+     * @return void
+     */
+    public function validateDataEntityKey($str, $filename = null)
```

Comment:
can this be repurposed as below for all camel case validations?
```
validateEntityInCamelCase($str, $type, $filename = null)
{
..
$message = "The ($type) {$str} should be PascalCase with an uppercase first letter.";
..
}
```

Reply:
fixed